games/rpg/src/lib.rs

```diff
@@ -0,0 +1,1483 @@
+                        .mint_non_fungible(&NonFungibleId::random(), new));
+                return new_bucket
+            }
+            else if which == 2 {
+                let mut item: Armor = item_bucket.non_fungibles()[0].data();
+                let item2: Armor = item_bucket.non_fungibles()[1].data();   
+                assert!(item.level == item2.level);            
+                assert!(item.id == item2.id);
+                assert!(item.part == item2.part);
+                item.level += dec!(1); 
+                item.health *= dec!("1.2");
+                item.defense *= dec!("1.2");
+                item.ability_scaling *= dec!("1.2");  
+                let new = Armor { id: item.id, part: item.part, level: item.level, health: item.health,
+                    defense: item.defense, weight: item.weight, ability: item.ability, ability_scaling: item.ability_scaling, version: self.version,
+                    };
+                self.system_vault.authorize(||item_bucket.burn());
+                let new_bucket = self.system_vault.authorize(||
+                    borrow_resource_manager!(self.item_nft)
+                        .mint_non_fungible(&NonFungibleId::random(), new));
+                return new_bucket
+            }
+            else {
+                let mut item: Accessory = item_bucket.non_fungibles()[0].data();
+                let item2: Accessory = item_bucket.non_fungibles()[1].data();  
+                assert!(item.level == item2.level);        
+                assert!(item.id == item2.id);
+                assert!(item.part == item2.part);
+                item.level += dec!(1); 
+                item.attack *= dec!("1.2");
+                item.magic *= dec!("1.2");
+                item.speed *= dec!("1.2");
+                item.ability_scaling *= dec!("1.2");  
+                let new = Accessory { id: item.id, part: item.part, level: item.level, attack: item.attack,
+                    magic: item.magic, speed: item.speed, weight: item.weight, ability: item.ability, ability_scaling: item.ability_scaling, version: self.version,
+                    };
+                self.system_vault.authorize(||item_bucket.burn());
+                let new_bucket = self.system_vault.authorize(||
+                    borrow_resource_manager!(self.item_nft)
+                        .mint_non_fungible(&NonFungibleId::random(), new));
+                return new_bucket
+            }              
+        }
+        // Creates weapons
+        pub fn create_weapon_1(&mut self, gold: Bucket, resource1: Bucket, resource2: Bucket, class: u8) -> Bucket {
+            assert!(gold.resource_address() == self.token_gold);
+            assert!(resource1.resource_address() == self.token_greavite);
+            assert!(resource2.resource_address() == self.token_wood);
+            let key_bucket: Bucket = self.system_vault.take(1);
+            let class_chosen = if class == 1 { String::from("Sword") }
```

Comment:
Why not use a `HashMap` for this which maps the class enum (which I commented about previously) to a tuple of values? Something like: `HashMap<ClassType, (String, Decimal, Decimal, Decimal, Decimal)>`.